### PR TITLE
[Storage Migration] V3 jobs

### PR DESCRIPTION
**Pull request by dannyjdev**
# Changes
* Updates V3 jobs to use `useManagedIdentity` and `clientId` parameters from the configuration files.
* Use MSI or SAS token for Jobs that uses both storages global and china in a single configuration.
* Addresses the following jobs:
  - `Cataog2DnxJob`
  - `Catalog2IconJob`
  - `Catalog2RegistrationJob`
  - `Db2CatalogJob`
  - `FixCatalogCachingJob`
  - `LightningJob`
  
Addresses: https://github.com/NuGet/Engineering/issues/5446

### CODE REVIEW

File: src/Ng/CommandHelpers.cs [hunk 1]
```diff
@@ -427,10 +430,21 @@ private static IBlobServiceClientFactory GetBlobServiceClient(
             IDictionary<string, string> argumentNameMap)
         {
             bool storageUseManagedIdentity = arguments.GetOrDefault(argumentNameMap[Arguments.StorageUseManagedIdentity], defaultValue: false);
-            if (storageUseManagedIdentity)
+            bool useManagedIdentity = storageUseManagedIdentity || arguments.GetOrDefault(argumentNameMap[Arguments.UseManagedIdentity], defaultValue: false);
+
+            var storageKeyValue = arguments.GetOrDefault<string>(argumentNameMap[Arguments.StorageKeyValue]);
+            var storageSasValue = arguments.GetOrDefault<string>(argumentNameMap[Arguments.StorageSasValue]);
+
+            var hasStorageKeyOrSas = !string.IsNullOrEmpty(storageKeyValue) || !string.IsNullOrEmpty(storageSasValue);
```
Comment (erdembayar) [inline src/Ng/CommandHelpers.cs:438]: Nit: use explicit boolean type.

File: src/Ng/CommandHelpers.cs [hunk 2]
```diff
@@ -427,10 +430,21 @@ private static IBlobServiceClientFactory GetBlobServiceClient(
...
+
+            if (useManagedIdentity && !hasStorageKeyOrSas)
```
Comment (erdembayar) [inline src/Ng/CommandHelpers.cs:440]: For sanity check should be throw exception if both `useManagedIdentity` and `hasStorageKeyOrSas` are enabled.

Comment (dannyjdev) [reply src/Ng/CommandHelpers.cs:440]: No, unfortunately there are jobs that have multiple storage in their config that access both global (with MSI) and china (with sas token), so if useManagedIdentity is defined we need to also check for SAS token in china storage case.

**Review by erdembayar — DISMISSED** [review]

File: src/Ng/CommandHelpers.cs (hunk 2 above)
Comment (erdembayar) [reply src/Ng/CommandHelpers.cs:440]: Could you add small comment about this situation so next time we know why we have this setup?

File: src/Ng/CommandHelpers.cs (hunk 1 above)
Comment (dannyjdev) [reply src/Ng/CommandHelpers.cs]: Addressed.

File: src/Ng/CommandHelpers.cs (hunk 2 above)
Comment (dannyjdev) [reply src/Ng/CommandHelpers.cs:442]: Added a comment for that.

**Review by erdembayar — DISMISSED** [review]

**Review by erdembayar — DISMISSED** [review]

File: src/NuGet.Jobs.Catalog2Registration/DependencyInjectionExtensions.cs [hunk 3]
```diff
@@ -57,7 +68,16 @@ private static void RegisterCursorStorage(ContainerBuilder containerBuilder)
                 {
                     var options = c.Resolve<IOptionsSnapshot<Catalog2RegistrationConfiguration>>();
 
-                    // workaround for https://github.com/Azure/azure-sdk-for-net/issues/44373
+                    if (options.Value.StorageUseManagedIdentity && !options.Value.HasSasToken)
+                    {
+                        var credential = new DefaultAzureCredential(
```
Comment (erdembayar) [inline src/NuGet.Jobs.Catalog2Registration/DependencyInjectionExtensions.cs:73]: @dannyjdev 
Should we replace this with `ManagedIdentityCredential` one too?

**Review by erdembayar — APPROVED** [review]

**Pull request merged**